content/blog/standard-tf-files.md

```diff
@@ -0,0 +1,414 @@
+
+## Supporting Configuration Files
+
+The previous files are extremely common across most TF projects we see and focus on your infrastructure. Let's look at other supporting types of files which connect your infrastructure to external services.
+
+### providers.tf: Managing Provider Configurations
+
+The providers.tf file centralizes all your TF provider configurations in one place. Providers are plugins that Terraform uses to interact with cloud platforms, services, and APIs. By isolating these configurations, you make it easier to understand which external services your module is interacting with.
+
+Here's an example providers.tf file:
+
+```hcl
+provider "aws" {
+  region = var.aws_region
+
+  assume_role {
+    role_arn = var.deployment_role_arn
+  }
+
+  default_tags {
+    tags = var.default_tags
+  }
+}
+
+provider "aws" {
+  alias  = "us-west-2"
+  region = "us-west-2"
+}
+```
+
+Provider configurations often handle authentication methods, region settings, and default parameters. Having these in a dedicated file makes it simple to update authentication approaches or other shared settings across your entire project.
+
+When multiple configurations of the same provider are needed, such as deploying resources across different AWS regions, the alias parameter distinguishes between them. These aliased providers can then be referenced in resource blocks throughout your configuration. In the example above, the second provider can be referenced as aws.us-west-2 in your resources.
+
+### versions.tf: Terraform and Provider Requirements
+
+While providers.tf configures how TF interacts with external services such as cloud providers, versions.tf defines what versions of TF and those providers are compatible with your code. This distinction helps manage compatibility at both the tool level and the API level.
+
+Here's an example versions.tf file:
+
+```hcl
+terraform {
+  required_version = ">= 1.3.0, < 2.0.0"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 4.16"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = ">= 3.4.3"
+    }
+  }
+}
+```
+
+The required_version argument ensures your configuration runs with a compatible TF version and fails if none are available. This prevents mysterious errors when teammates use different versions. At Masterpoint, for child modules (reusable modules that are called by other TF configurations), we specify a minimum version that supports the features we need.
+
+The required_providers block serves two important functions:
+
+- where TF should download providers from
+- what version constraints apply
+
+The source attribute is particularly important in organizations that use private provider registries or forks of official providers.
+
+Want to know all about versioning in Terraform and OpenTofu? Read our definitive article on that topic: [The Ultimate Terraform Versioning Guide](/blog/ultimate-terraform-versioning-guide). 
```

Comment:
@gberenice if we change the name / slug of your article then we need to update this! Pinging you so you can help me remember 😁